doc/Programming-guide/Key-platform-features/Endpoint-profiles/index.md

```diff
@@ -10,38 +10,49 @@ sort_idx: 20
-  
-## Client-side endpoint profile ##
-
-The client-side endpoint profile is initially generated at the stage of a new endpoint registration. Prior to that, the client-side endpoint profile values should be 
-specified by the client developer using the Kaa endpoint SDK. The client-side endpoint profile can be also updated at run time using SDK API call. In this case, SDK executes 
-profile update request and endpoint membership in the endpoint groups is re-evaluated and updated to match the new endpoint profile.
-The client-side endpoint profile is unidirectionally synchronized, and thus should not be considered as a means to temporarily store endpoint data in the Kaa cluster. 
-There is no way for the endpoint to retrieve the profile information back from the Kaa cluster. At its start, the application must fill in the current endpoint profile 
-with the up-to-date data and execute corresponding SDK API calls. The endpoint SDK does not persist the profile information over the endpoint reboots. However, 
-it detects profile data changes and submits the new data to the Kaa cluster as a profile update.
+[Endpoint profile]({{root_url}}Glossary/#endpoint-profile-client-side-server-side) (EP profile) is a structured set of data that describes specific characteristics of an [endpoint]({{root_url}}Glossary/#endpoint-ep).
+
+The structure of both client-side and server-side endpoint profile is a customizable structured data set that describes specific characteristics of the endpoint.
+The profiles are used to attribute endpoints to [endpoint groups]({{root_url}}Glossary/#endpoint-group).
+Every endpoint profile comprises the *client-side*, *server-side* and *system* parts.
+
+The structure of both client-side and server-side parts of an endpoint profile are defined by [application]({{root_url}}Glossary/#kaa-application) developer using the [Apache Avro schema](http://avro.apache.org/docs/current/spec.html#schemas) format.
+Application developer can reuse and share certain data structures using the [common type library (CTL)]({{root_url}}Glossary/#common-type-library-ctl).
+
+The client-side structure is used during the SDK generation.
+Therefore, if you make changes to the client-side structure, you need to re-generate your SDK.
+For more information, see [Endpoint SDK]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs).
+
+The structure of the system part is identical across the applications and is used by [Kaa platform]({{root_url}}Glossary/#kaa-platform) to perform internal functions.
+Both client-side and server-side profile schemas are maintained within the corresponding application, with its own version that distinguishes it from the previous schemas.
+Multiple schema versions and the corresponding endpoint profiles can coexist within a single application.
+
+## Client-side EP profile
+
+Initial values for the client-side part are specified by the the client developer using data schemas for the endpoint SDK.
```

Comment:
I know what this is supposed to mean, but even I struggle to decrypt. Do you understand the meaning? Nobody will.